lib/engine/game/g_1828.rb

```diff
@@ -221,6 +230,131 @@ def upgrades_to?(from, to, special = false)
         super
       end
 
+      def merge_candidates(player, corporation)
+        return [] if !player || !corporation
+        return [] if corporation.system?
+
+        @corporations.select do |candidate|
+          next if candidate == corporation ||
+                  candidate.system? ||
+                  !candidate.ipoed ||
+                  (corporation.owner != player && candidate.owner != player) ||
+                  candidate.operated? != corporation.operated? ||
+                  (!candidate.floated? && !corporation.floated?)
+
+          # account for another player having 5+ shares
+          @players.any? do |p|
+            num_shares = p.num_shares_of(candidate) + p.num_shares_of(corporation)
+            num_shares >= 6 ||
+              (num_shares == 5 && !sold_this_round?(p, candidate) && !sold_this_round?(p, corporation))
+          end
+        end
+      end
+
+      def sold_this_round?(entity, corporation)
+        return false unless @round.players_sold
+
+        @round.players_sold[entity][corporation]
+      end
+
+      def create_system(corporations)
+        system_data = CORPORATIONS.find { |c| c['sym'] == corporations.first.id }.dup
+        system_data['sym'] = corporations.map(&:name).join('-')
+        system_data['tokens'] = []
+        system = init_corporation(@stock_market, system_data)
+        system.extend(Engine::G1828::System)
```

Comment:
why do you need to extend it like this?, why can't you just do System.new?

Reply:
I felt like putting the the corporation initialization logic into the 1828 game class was a fragile design. If anyone ever changed it, 1828 would likely break. This way, I'm decorating my additions and leaving everything else alone. I actually thought this was kind of elegant. What do you think?

Reply:
what do you mean? why can't there just be an 1828 specific corporation?

Reply:
Are you suggesting inherit from init_corporations and call System.new from there? That would be fine. What I don't want to do is duplicate the init logic.

Reply:
what do you mean inherit from init_corporations? how do you inherit a method?

Reply:
Bad language -- it was late last night. I meant override.

Reply:
Refactored.